### PR TITLE
faust: make faust2sc script available

### PR DESCRIPTION
**Pull request by olafklingt**
###### Description of changes

faust2sc is a script that is part of faust to generate SuperCollider UGens this pr makes it availble via nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/faust/faust2supercollider.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  postFixup = ''
+    # export parts of the build environment
+    for script in "$out"/bin/*; do
+       wrapProgram "$script" \
+        --set SUPERCOLLIDER_HEADERS "${supercollider}/include/SuperCollider/" \
+         --prefix PATH : "$PATH"
+     done
+   '';
+})
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faust/faust2supercollider.nix:26]:
```suggestion
{ faust
, supercollider
, makeWrapper
, stdenv
}@args:

stdenv.mkDerivation (faust.faust2ApplBase (args // {
  baseName = "faust2supercollider";
})) // {
  nativeBuildInputs = [ makeWrapper ];

  propagatedBuildInputs = [ faust supercollider ];

  postInstall = ''
  '';

  postFixup = ''
    # export parts of the build environment
    for script in "$out"/bin/*; do
      wrapProgram "$script" \
        --set SUPERCOLLIDER_HEADERS "${supercollider}/include/SuperCollider/" \
        --prefix PATH : "$PATH"
    done
  '';
}
```
meta is missing

**Review by SuperSandro2000 — COMMENTED** [review]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

File: pkgs/applications/audio/faust/faust2supercollider.nix (hunk 1 above)
Comment (olafklingt) [reply pkgs/applications/audio/faust/faust2supercollider.nix:26]: what is meta?

Comment (olafklingt) [reply pkgs/applications/audio/faust/faust2supercollider.nix]: removing the brackets passes the main attribute set not to mkDerivation

Comment (olafklingt) [reply pkgs/applications/audio/faust/faust2supercollider.nix]: > meta is missing

I thought it is sufficient to use the meta tags from [faust](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/audio/faust/faust2.nix) like the other files in this folder.

**Comment by olafklingt** [comment]
@magnetophon i had to add some path arguments to the faust2supercollider that I would expect to be already setup by faust itself.
The problem doesn't appear if i Faust within my setup, but when i use faust2supercollider within a flake i had to use those two path arguments:
```
faust -I ${pkgs.faust}/share/faust -A ${pkgs.faust}/include
```
(`-pathlist` shows the paths visible to Faust)
I made faust2supercollider work. But I would suggest to wrap Faust itself having these two arguments.

**Comment by olafklingt** [comment]
I realized there is a better build script for SuperCollider UGens provided by faust. The suggestion to wrap faust is still relevant.

**Comment by magnetophon** [comment]
@olafklingt Thanks for making this!
Unfortunately I only used supercollider briefly, a long time ago and I have never used any flakes.
I'm happy to test this, but you're going to have to give me some guidance!

**Comment by olafklingt** [comment]
hi @magnetophon  i am also still learning how to use flakes but it was a good exercise to create a test environment test a faust generated UGen you can use this repository as described in the readme to test this PR https://github.com/olafklingt/faust2sc_test 

if it doesn't work please tell me here or on matrix

**Comment by magnetophon** [comment]
@olafklingt Thanks for the test repo!
Unfortunately, when I run ``nix run github.com:olafklingt/faust2sc_test#``, I get: 
``error: input 'github.com:olafklingt/faust2sc_test#' is unsupported``.

**Comment by olafklingt** [comment]
@magnetophon ah sorry `nix run github:olafklingt/faust2sc_test#`

**Comment by magnetophon** [comment]
Cool, that works. 
Except that's not a 200Hz tone, but OK.  :grin:

**Review by magnetophon — APPROVED** [review]

**Comment by olafklingt** [comment]
what needs to be done to merge this?

**Pull request merged**